fem/restriction.cpp

```diff
@@ -837,10 +837,12 @@ void ConformingFaceRestriction::SetFaceDofsScatterIndices(
    const int* elem_map = e2dTable.GetJ();
    const int elem_index = face.element[0].index;
 
+   const int * h_vol_dof_map = vol_dof_map.HostRead();
+
    for (int face_dof = 0; face_dof < face_dofs; ++face_dof)
    {
       const int lex_volume_dof = face_map[face_dof];
-      const int s_volume_dof = vol_dof_map[lex_volume_dof]; // signed
+      const int s_volume_dof = h_vol_dof_map[lex_volume_dof]; // signed
```

Comment:
I suspect the "issue" in debug mode was that here we're calling the non-const version of `operator[]`. In that case the debug check will generate an error if the memory is not valid only on host. Since we're actually just reading the host memory here, we can force the call to be to the const version of `operator[]` by using `AsConst(vol_dof_map)[lex_volume_dof]`. In the const version, the debug check only makes sure the host pointer is valid and does not care (since we're only reading on host) about the device validity flag.

Using `AsConst`, there's no need to define the additional host pointer, `h_vol_dof_map`.

Reply:
Thanks for pointing that out, just pushed up the adjustments